File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> mpp;
        for (int i = 0; i < nums.size(); i++) {
            mpp[nums[i]]++;
        }
        vector<pair<int, int>> arr;
        for (auto mp : mpp) {
            arr.push_back({mp.second, mp.first});
        }
        sort(arr.rbegin(), arr.rend());
        vector<int> res;
        for (int i = 0; i < k; ++i) {
            res.push_back(arr[i].second);
        }
        return res;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function (nums, k) {
    const freqMap = new Map()
    const result = []
    for (let num of nums) {
        freqMap.set(num, (freqMap.get(num) || 0) + 1)
    }
    const sortedMap =
        [...freqMap].sort((a, b) => b[1] - a[1])
    console.log("sortedMap", sortedMap)

    for (let i = 0; i < k; i++) {
        result.push(sortedMap[i][0])
    }



    return result;
};